dapp/src/components/dialogs/SetPermissionsDialog.tsx

```diff
@@ -0,0 +1,226 @@
+    DialogBody,
+    DialogContent,
+    DialogPosition,
+    Header,
+    InfoBox,
+    InfoBoxStyle,
+    InfoBoxType,
+    LoadingIndicator,
+} from '@iota/apps-ui-kit';
+import { useCurrentAccount, useIotaClient, useSignAndExecuteTransaction } from '@iota/dapp-kit';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { ChangeEvent, useState } from 'react';
+import toast from 'react-hot-toast';
+
+import { NameRecordData, queryKey, useNameRecord, useRegistrationNfts } from '@/hooks';
+import { NameUpdate, useUpdateNameTransaction } from '@/hooks/useUpdateNameTransaction';
+import { RegistrationNft } from '@/lib/interfaces';
+import { getNamePermissions, getParentObject, isNameRecordExpired } from '@/lib/utils/names';
+
+function buildPermissionsUpdate({
+    name,
+    ownedNames,
+    ownedSubnames,
+    editIsAllowingRenew,
+    editIsAllowSubnames,
+    namePermissions,
+}: {
+    name: string;
+    ownedNames: RegistrationNft[];
+    ownedSubnames: RegistrationNft[];
+    editIsAllowingRenew: boolean;
+    editIsAllowSubnames: boolean;
+    namePermissions?: ReturnType<typeof getNamePermissions>;
+}) {
+    const updates: NameUpdate[] = [];
+    const nftId = getParentObject(ownedNames ?? [], ownedSubnames ?? [], name)?.id;
+    if (
+        nftId &&
+        (editIsAllowingRenew !== namePermissions?.allowTimeExtension ||
+            editIsAllowSubnames !== namePermissions?.allowChildCreation)
+    ) {
+        updates.push({
+            type: 'edit-setup',
+            name,
+            parentNftId: nftId,
+            allowChildCreation: editIsAllowSubnames,
+            allowTimeExtension: editIsAllowingRenew,
+        });
+    }
+
+    return {
+        updates,
+    };
+}
+
+type CreateSubnameProps = {
+    name: string;
+    setOpen: (bool: boolean) => void;
+};
+
+export function SetPermissionsDialog({ name, setOpen }: CreateSubnameProps) {
+    const queryClient = useQueryClient();
+    const iotaClient = useIotaClient();
+    const account = useCurrentAccount();
+    const { data: ownedNames = [] } = useRegistrationNfts('name');
+    const { data: ownedSubnames = [] } = useRegistrationNfts('subname');
+    const { data: nameRecordData, isLoading: isNameRecordLoading } = useNameRecord(name);
+    // We are sure that only owned names are passed here
+    const nameRecord = nameRecordData as
+        | Extract<NameRecordData, { type: 'unavailable' }>
+        | undefined;
+
+    const namePermissions = nameRecord?.nameRecord
+        ? getNamePermissions(nameRecord.nameRecord)
+        : undefined;
+
+    const isExpired = nameRecord?.nameRecord ? isNameRecordExpired(nameRecord?.nameRecord) : false;
+
+    // Editable values
+    const [editIsAllowingRenew, setEditIsAllowingRenew] = useState<boolean>(
+        namePermissions?.allowTimeExtension ?? false,
+    );
+    const [editIsAllowSubnames, setEditIsAllowSubnames] = useState<boolean>(
+        namePermissions?.allowChildCreation ?? false,
+    );
+
```

Comment:
This is an assumption I would not rely on. The name record might or not be present when this dialog runs. I suggest you add the sync that was already before:
```suggestion

    // Sync permissions
    useEffect(() => {
        if (namePermissions) {
            setEditIsAllowingRenew(namePermissions.allowTimeExtension);
            setEditIsAllowSubnames(namePermissions.allowChildCreation);
        }
    }, [namePermissions?.allowChildCreation, namePermissions?.allowTimeExtension]);

```